docs/devguide/fields-yml.asciidoc

```diff
@@ -0,0 +1,116 @@
+`enabled`:: Whether or not the field is enabled.
+`analyzer`:: Which analyzer to use when indexing.
+`search_analyzer`:: Which analyzer to use when searching.
+`norms`:: Applies to `text` and `keyword` fields. Default is `false`.
+`dynamic`:: Dynamic field control. Can be one of `true` (default), `false` or `strict`.
+`index`:: Whether or not the field should be indexed.
+`doc_values`:: Whether or not the field should have doc values generated. See docs.
+`copy_to`:: Which field to copy the field value into.
+`ignore_above`:: When this property value is missing or is `0`, it receives the `libbeat` default value of `1024`. If the value is `-1`, the Elasticsearch default value will be applied. Any other specified value will be applied as-is.
+
+If we wanted to be able to search over both name fields in one go, we could copy the `surname` and `forenames` fields into the `full_name` field at index time using the `copy_to` mapping parameter:
+[source,yaml]
+----------------------------------------------------------------------
+- key: mybeat
+  title: mybeat
+  description: These are the fields used by mybeat.
+  fields:
+    - name: surname
+      type: keyword
+      required: true
+      copy_to: full_name <1>
+      description: >
+        The surname.
+    - name: forenames
+      type: keyword
+      required: true
+      copy_to: full_name <2>
+      description: >
+        The forenames.
+    - name: full_name
+      type: text
+      required: false
+      description: >
+        The surname and forenames combined into one field for easy searchability.
+----------------------------------------------------------------------
+<1> Copy the value of `surname` into `full_name`
+<2> Copy the value of `forenames` into `full_name`
+
+There are also some Kibana-specific properties, not detailed here. These are: `analyzed`, `count`, `searchable`, `aggregatable`, `script`. Kibana parameters can also be described using the `pattern`, `input_format`, `output_format`, `output_precision`, `label_template`, `url_template` and `open_link_in_current_tab` parameters.
+
+==== Defining text multi-fields
+There are various options that can be applied when using text fields. A simple text field using the default analyzer can be defined without any other options, as in the example above.
+
+When using `text` mappings, if you need to keep the original keyword value, for instance to use in aggregations or ordering, you can apply a multi-field mapping. We can use this in our mapping for the `surname` and `forenames` fields like this:
+[source,yaml]
+----------------------------------------------------------------------
+- key: mybeat
+  title: mybeat
+  description: These are the fields used by mybeat.
+  fields:
+    - name: surname
+      type: keyword
+      required: true
+      copy_to: full_name
+      description: >
+        The surname.
+      multi_fields: <1>
```

Comment:
Both `surname` and `forenames` here are themselves `keyword` types, which would make this redundant, right? The description above also describes using this on text fields, not keywords. Perhaps this section should use a different example field that would more naturally be of type text.

Reply:
@bestpath-gb Will you be able to update the example as suggested here? I appreciate the effort that you put into creating this PR and would like to help you get this merged.

Reply:
Thanks for the feedback. Sorry for the delay in getting back to this. I'm hoping to review and update the examples this week.

Reply:
That was a great catch, @faec . I've updated the examples based on the Elasticsearch reference for those mapping parameters.